src/io/eval.rs

```diff
@@ -238,13 +238,17 @@ where
                 encode.plaintext.clone()
             };
             log_mem(format!("plaintext at {}, {} computed", level, j));
+            // C_att, xL: new encoding of next level
             let new_encode: BggEncoding<M> =
                 BggEncoding::new(new_vec, pub_key_level[j].clone(), plaintext);
             log_mem(format!("new_encode at {}, {} computed", level, j));
             new_encodings.push(new_encode);
         }
+        // s_xL: input dependent secret key
         ps.push(p.clone());
+        // A_xL: input independent public key
         pub_key_cur = pub_key_level;
+        // c_att_xL: BGG+ encoding of level
```

Comment:
Similarly to what I mentioned in the `obf.rs` file. I think that, in the comments, we should treate the encodings and the public keys as a unique object, without separating between c_att and c_t.